packages/back-end/src/models/FactMetricModel.ts

```diff
@@ -217,6 +217,27 @@ export class FactMetricModel extends BaseClass {
       }
     }
 
+    // validate column
+    const metricSupportsDistinctDates =
+      data.metricType === "mean" ||
+      data.metricType === "ratio" ||
+      (data.metricType === "quantile" &&
+        data.quantileSettings?.type === "unit");
+    if (data.numerator.column === "$$distinctDates") {
+      if (!metricSupportsDistinctDates) {
+        throw new Error(
+          "$$distinctDates is only supported for mean, ratio, and quantile metrics",
+        );
+      }
+    }
+    if (data.denominator?.column === "$$distinctDates") {
+      if (!metricSupportsDistinctDates) {
+        throw new Error(
+          "$$distinctDates is only supported for mean, ratio, and quantile metrics",
+        );
```

Comment:
Is this check needed?  If there's a denominator, that means it is a ratio metric and supported already.  I guess this is to future-proof things if we add more restrictive conditions to the supports logic above?